Source/Core/DolphinQt2/Config/GeckoWidget.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QWidget>
+
+#include "Common/IniFile.h"
+#include "Core/GeckoCode.h"
```

Comment:
This can be a forward declaration. You can forward declare the type in a `std::vector` (as well as `std::list` and `std::forward_list`) since [N4510](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4510.html) was voted into C++17.